### PR TITLE
Update module Ubuntu 14.04 default to official repository setup

### PR DESCRIPTION
**Pull request by jk2l**
Fix for Ubuntu 14.04 Php installation:

Summary:
* The currents rule only work for PPA. Using 14.04 Official Ubuntu repository does not match any of the rules. 
* Removed 14.10 from metadata. It is in EOL and thing can randomly break. 
* Removed 12.04 from metadata. Default installation is 5.3, this module won't support any of it

### CODE REVIEW

**Comment by jk2l** [comment]
The module tested against both 14.04 and 16.04 ubuntu. But our test case will need to patch according to it as almost everywhere hardcoded for php5-

File: manifests/dev.pp [hunk 1]
```diff
@@ -26,7 +26,8 @@
     default   => $package,
   }
 
-  if $::operatingsystem == 'Ubuntu' {
+  # Default PHP come with xml module and no seperate package for it
+  if $::operatingsystem == 'Ubuntu' and $::operatingsystemrelease >= '16.04'  {
```
Comment (bastelfreak) [inline manifests/dev.pp]: can you use versioncompare here instead of >=?

File: manifests/pear.pp [hunk 2]
```diff
@@ -43,7 +43,8 @@
   validate_string($ensure)
   validate_string($package_name)
 
-  if $::operatingsystem == 'Ubuntu' {
+  # Default PHP come with xml module and no seperate package for it
+  if $::operatingsystem == 'Ubuntu' and $::operatingsystemrelease >= '16.04' {
```
Comment (bastelfreak) [inline manifests/pear.pp]: also versioncompare here

File: spec/classes/php_fpm_spec.rb [hunk 3]
```diff
@@ -9,16 +9,9 @@
...
-          when 'Ubuntu'
-            let(:params) { { package: 'php5.6-fpm', ensure: 'latest' } }
-            it { is_expected.to contain_package('php5.6-fpm').with_ensure('latest') }
-            it { is_expected.to contain_service('php5.6-fpm').with_ensure('running') }
-          when 'Debian'
-            let(:params) { { package: 'php5-fpm', ensure: 'latest' } }
-            it { is_expected.to contain_package('php5-fpm').with_ensure('latest') }
-            it { is_expected.to contain_service('php5-fpm').with_ensure('running') }
-          end
+          let(:params) { { package: 'php5-fpm', ensure: 'latest' } }
```
Comment (bastelfreak) [inline spec/classes/php_fpm_spec.rb]: can you migrate the let block to multiline do/end?

File: spec/defines/extension_spec.rb [hunk 4]
```diff
@@ -6,17 +6,15 @@
       let :facts do
         facts
       end
-      let(:pre_condition) { 'include php::params' }
+      let(:pre_condition) { '
```
Comment (bastelfreak) [inline spec/defines/extension_spec.rb]: replace {} with do/end

**Comment by bastelfreak** [comment]
versioncmp() is documented here https://docs.puppet.com/puppet/latest/reference/function.html#versioncmp

**Pull request merged**